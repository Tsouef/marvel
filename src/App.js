import React from 'react';
import { Router, Route } from 'react-router-dom';

import Landing from './components/Landing/Landing';
import Header from './components/Header/Header';
import Notification from './components/Notification/Notification';
import ComicsList from './components/Comics/ComicsList';
import ComicsFilter from './components/Comics/ComicsFilter';
import Comic from './components/Comics/Comic';
import CharactersList from './components/Characters/CharactersList';
import CharactersFilter from './components/Characters/CharactersFilter';
import Character from './components/Characters/Character';
import Favorites from './components/Favorites/Favorites';
import CreateForm from './components/Form/CreateForm';
import Auth from './components/Auth/Auth';
import PrivateRoute from './PrivateRoute';
import history from './history';

const App = () => {
  return (
    <Router history={history}>
      <div className="container" style={{ position: 'relative' }}>
        <Route path="/" component={Header} />
        <Route path="/" component={Notification} />
        <Route exact path="/login" component={Auth} />
        <PrivateRoute exact path="/" component={Landing} />
        <PrivateRoute exact path="/comics" component={ComicsList} />
        <PrivateRoute exact path="/comics/id/:id" component={Comic} />
        <PrivateRoute
          exact
          path="/comics/letter/:letter"
          component={ComicsFilter}
        />
        <PrivateRoute exact path="/characters" component={CharactersList} />
        <PrivateRoute exact path="/characters/id/:id" component={Character} />
        <PrivateRoute
          exact
          path="/characters/letter/:letter"
          component={CharactersFilter}
        />
        <PrivateRoute exact path="/favorites" component={Favorites} />
        <PrivateRoute exact path="/add" component={CreateForm} />
      </div>
    </Router>
  );
};

export default App;
