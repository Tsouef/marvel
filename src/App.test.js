import App from './App';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<App />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
