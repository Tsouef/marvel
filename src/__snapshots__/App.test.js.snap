// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render with default props 1`] = `
<BrowserRouter>
  <div
    className="container"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/comics"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/comics/id/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/comics/letter/:letter"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/characters"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/characters/id/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/characters/letter/:letter"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/favorites"
    />
  </div>
</BrowserRouter>
`;
